src/ai/backend/test/templates/template.py

```diff
@@ -48,33 +78,44 @@ async def run_test(self, exporter: TestExporter) -> None:
 @final
 class BasicTestTemplate(TestTemplate):
     _testcode: TestCode
+    _wrapper_templates: list["WrapperTestTemplateProtocol"]
 
-    def __init__(self, testcode: TestCode) -> None:
+    def __init__(
+        self, testcode: TestCode, wrapper_templates: list["WrapperTestTemplateProtocol"] = []
+    ) -> None:
```

Comment:
It seems better to use `abstractmethod` rather than trying to use the constructor, since the constructors may vary for each class.